src/modules/job-info/lookup.c

```diff
@@ -289,9 +283,25 @@ static void info_lookup_continuation (flux_future_t *fall, void *arg)
 
         if ((l->flags & FLUX_JOB_LOOKUP_CURRENT)
             && streq (keystr, "R")) {
-            if (lookup_current (l, fall, keystr, s, &current_value) < 0)
+            json_t *current_object = NULL;
+            if (lookup_current (l, fall, keystr, s, &current_object) < 0)
                 goto error;
-            s = current_value;
+            /* tiny optimization, to skip unnecessary json_dumps() and
+             * fallthrough to code below
```

Comment:
I'd be fine dropping this optimization for now and only address performance if it becomes an issue.